File: TheQuizApp/app/src/main/java/com/example/thequizapp/viewmodel/QuizViewModel.java
package com.example.thequizapp.viewmodel;

import android.util.Log;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.example.thequizapp.model.QuestionList;
import com.example.thequizapp.repository.QuizRepository;

public class QuizViewModel extends ViewModel {

    QuizRepository repository = new QuizRepository();

    LiveData<QuestionList> questionListLiveData;

    public QuizViewModel() {
        questionListLiveData = repository.getQuestionsFromAPI();

    }

    public LiveData<QuestionList> getQuestionListLiveData() {
        Log.v("MyActivity", questionListLiveData.toString());
        return questionListLiveData;
    }
}

File: KotlinQuizApp/app/src/main/java/com/example/kotlinquizapp/viewmodel/QuizViewModel.kt
package com.example.kotlinquizapp.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.example.kotlinquizapp.model.QuestionsList
import com.example.kotlinquizapp.repository.QuizRepository

class QuizViewModel: ViewModel() {

    var repository: QuizRepository = QuizRepository()

    lateinit var questionsLiveData: LiveData<QuestionsList>

    init {
        questionsLiveData = repository.getQuestionsFromAPI()

    }

    fun getQuestionsFromLiveData(): LiveData<QuestionsList> {
        return questionsLiveData
    }

}